File: ruby/src/sorts/selection_sort.rb
module SelectionSort
    def SelectionSort.sort(arr)
        # copy the input array
        out = arr.map { |x| x }

        for i in (0...out.length)
            min = i
            for j in (i...out.length)
                if out[j] < out[min]
                    min = j
                end
            end
            out[i], out[min] = out[min], out[i]
        end

        out
    end
end

File: go/src/sorts/selectionsort.go
package sorts

func SelectionSort(arr []int) ([]int) {
  sorted := make([]int, len(arr))
  copy(sorted, arr)

  for i := 0; i < len(sorted); i++ {
    min := i

    for j := i; j < len(sorted); j++ {
      if sorted[j] < sorted[min] {
        min = j
      }
    }

    sorted[i], sorted[min] = sorted[min], sorted[i]
  }

  return sorted
}
